### DISCUSSION TITLE
Don't store switcher value / always start with "no" in switcher when opening the editor

### DISCUSSION THREAD

**Discussion by m1ga** [elementor/elementor#25541] [Developers Community]
### Start a discussion

Hi,

I'm registering a control in the page settings using
```php
add_action('elementor/documents/register_controls', [ $this, 'custom_contol' ]);
```
and in there I have a simple switch
```php
$controls_stack->add_control(
                'item_show',
                [
                      'label' => esc_html__('Show item', 'textdomain'),
                      'type' => \Elementor\Controls_Manager::SWITCHER,
                      'label_on' => esc_html__('Show', 'textdomain'),
                      'label_off' => esc_html__('Hide', 'textdomain'),
                      'return_value' => 'yes',
                      'default' => 'no',
                      ]
                  ]
            );
```

How can I make it so it will always start with "no" when I open the editor. Currently it saves the state, so if I switch it on and save my page, reopen it again it will still be "yes". I always want it to start with "no".

**Comment by perks98** [comment]
@m1ga based on the link below, have you created `get_default_value()` function when you created the control and set the default value there?

Link: [https://developers.elementor.com/docs/controls/control-values/](https://developers.elementor.com/docs/controls/control-values/)

**Reply by m1ga** [reply]
In this case I can't set that as I'm injection my control into the document settings, so I'm not extending ` \Elementor\Widget_Base`. Also if I understand the docs right: it will just set the default value (which I do set in the control already) but once I change it and save it it will use the saved value and not the default. But I want it to not store the value or override it with `false` even if you save `true`
